### PR TITLE
Fixes for Issue#18 (More descriptive error messages in the UI)

### PR DESCRIPTION
**Pull request by gagansk**
To fix: https://github.com/FNNDSC/ChRIS_store_ui/issues/18

This is my first attempt to learn React and contribute to the ChRIS project. 

I have attached screenshots in the comments of the issue. Let me know if you need any changes.

### CODE REVIEW

**Comment by jdtzmn** [comment]
@gagansk lgtm. Ideally, the UI would reflect the server's response to a "create plugin" request, but this appears to be a step in the right direction.

File: src/components/CreatePlugin/CreatePlugin.js [hunk 1]
```diff
@@ -181,12 +181,39 @@ class CreatePlugin extends Component {
       pluginRepresentation,
     } = this.state;
 
+    // Build error message
+    let errorMsg = '';
+    // Track the number of missing fields in the submission form
+    let errorCount = 4;
```
Comment (jdtzmn) [inline src/components/CreatePlugin/CreatePlugin.js]: The `errorCount` should start at 0 and count up, but I would instead recommend using an array to hold all of the errors and then construct a human-readable message using 'if' blocks.

Look at the PR comment if you need inspiration.

File: src/components/CreatePlugin/CreatePlugin.js [hunk 2]
```diff
@@ -181,12 +181,39 @@ class CreatePlugin extends Component {
...
     if (!(
       pluginName.trim() && pluginImage.trim() &&
       pluginRepo.trim() && pluginRepresentation &&
       Object.keys(pluginRepresentation).length > 0
     )) {
-      return this.handleError('All fields are required.');
+      // Checks for individual field completion
+      if (!pluginName.trim()) {
+        errorMsg += ' Please enter the plugin name.';
+        errorCount -= 1;
```
Comment (jdtzmn) [inline src/components/CreatePlugin/CreatePlugin.js]: The problem with concatenating these messages is that the message will not come out to be human-readable:

`Please enter the Docker Image. Please enter the Public Repo.`

File: src/components/CreatePlugin/CreatePlugin.js [hunk 3]
```diff
@@ -181,12 +181,39 @@ class CreatePlugin extends Component {
...
+      }
+      if (!pluginImage.trim()) {
+        errorMsg += ' Please enter the Docker Image.';
+        errorCount -= 1;
+      }
+      if (!pluginRepo.trim()) {
+        errorMsg += ' Please enter the Piblic Repo.';
```
Comment (jdtzmn) [inline src/components/CreatePlugin/CreatePlugin.js]: Small typo. We all make them 😉.

File: src/components/CreatePlugin/CreatePlugin.js [hunk 4]
```diff
@@ -181,12 +181,39 @@ class CreatePlugin extends Component {
...
+        errorCount -= 1;
+      }
+      if (!Object.keys(pluginRepresentation).length > 0) {
+        errorMsg += ' Please upload the plugin Representation.';
```
Comment (jdtzmn) [inline src/components/CreatePlugin/CreatePlugin.js]: Why is the R in Representation capitalized, but not the p in plugin? Use standard case for messages whenever possible.

File: src/components/CreatePlugin/CreatePlugin.js (hunk 1 above)
Comment (gagansk) [reply src/components/CreatePlugin/CreatePlugin.js]: I will be using an array as suggested.

File: src/components/CreatePlugin/CreatePlugin.js (hunk 2 above)
Comment (gagansk) [reply src/components/CreatePlugin/CreatePlugin.js]: I will make changes for the message to be human-readable.

File: src/components/CreatePlugin/CreatePlugin.js (hunk 3 above)
Comment (gagansk) [reply src/components/CreatePlugin/CreatePlugin.js]: Yes, we all do :)

File: src/components/CreatePlugin/CreatePlugin.js (hunk 4 above)
Comment (gagansk) [reply src/components/CreatePlugin/CreatePlugin.js]: This has been taken care in the latest commit.

**Comment by gagansk** [comment]
Thank you for your feedback. I have made the changes suggested.

**Comment by gagansk** [comment]
Closing this pull request and will open a new pull request for a non-master branch as suggested by @jcaianirh

**Pull request closed without merging**